wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -592,18 +607,21 @@ private[bitcoins] trait TransactionProcessing extends WalletLogger {
     }
 
     val nested = for {
-      addressDbWithOutput <- safeDatabase.run(addressDbWithOutputA)
+      addressDbWithOutput <- addressDbWithOutputA
     } yield {
-      val outputsVec = addressDbWithOutput.map { case (addressDb, out) =>
-        require(addressDb.scriptPubKey == out.output.scriptPubKey)
-        processReceivedUtxo(
-          transaction,
-          out.index,
-          blockHashOpt,
-          addressDb
-        )
+      val outputsVec: Vector[
+        DBIOAction[SpendingInfoDb, NoStream, Effect.Read with Effect.Write]] = {
+        addressDbWithOutput.map { case (addressDb, out) =>
+          require(addressDb.scriptPubKey == out.output.scriptPubKey)
+          processReceivedUtxo(
+            transaction,
+            out.index,
+            blockHashOpt,
+            addressDb
+          )
+        }
       }
-      Future.sequence(outputsVec)
+      slick.dbio.DBIOAction.sequence(outputsVec)
     }
 
     nested.flatten
```

Comment:
instead of having to do this .flatten could you just put the `DBIO.sequence(outputsVec)` in the `for` statement